libarchive/write.py

```diff
@@ -73,6 +73,47 @@ def add_files(self, *paths):
                         write_finish_entry(write_p)
                         entry_clear(entry_p)
 
+    def add_entry_from_memory(self, entry_path, entry_size, entry_data):
```

Comment:
@Changaco changes made, but I want to separate `_create_entry_archive` and `_close_entry_archive`. In my own code I subclass `ArchiveWrite` and rewrite body in `add_file_from_memory` without need for changing `_create_entry_archive` and `_close_entry_archive`.